istioctl/pkg/describe/describe.go

```diff
@@ -938,68 +963,105 @@ func printIngressInfo(
 		fmt.Fprintf(writer, "Skipping Gateway information (no ingress gateway pods)\n")
 		return nil
 	}
-	pod := pods.Items[0]
 
-	// Currently no support for non-standard gateways selecting non ingressgateway pods
-	ingressSvcs, err := kubeClient.CoreV1().Services(istioNamespace).List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return multierror.Prefix(err, "Could not find ingress gateway service")
-	}
-	filteredIngressSvcs := []corev1.Service{}
-	for _, svc := range ingressSvcs.Items {
-		if v, ok := svc.Spec.Selector["istio"]; ok && v == "ingressgateway" {
-			filteredIngressSvcs = append(filteredIngressSvcs, svc)
+	haveGateways := false
+	findVss := map[string]bool{}
+	for _, pod := range pods.Items {
```

Comment:
I think this can be improved with some pre-processing, like mapping the relationships between svcs & pods, vs & gateways, etc... the current logic is not quite scalable and the readability is not good with many loops.

Reply:
yes, that's a little inefficient